tsconfig.json

```diff
@@ -1,5 +1,37 @@
 {
   "compilerOptions": {
-    "esModuleInterop": true
-  }
+    /* Basic Options */
+    "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */,
```

Comment:
@fb55 sorry for the CC on an old PR, but I didn't want to make a new issue just for this. Is there any reason you didn't go with `ES2017` or whatever is equivalent for Node.js 10, which is the oldest supported Node.js version?

Reply:
Ah, NVM I see now that the package.json engines is set to `>=6`. On CI Node.js >= 10 is tested hence why I thought this could be improved. IMHO, these two should be in sync, especially since Node.js 6 is ancient...

Reply:
Jest doesn't support Node 6, which is why Node 6 isn't part of the CI anymore.

I am still a bit hesitant to bump the Node version, primarily due to the discussion in #1585. But the mismatch between CI and `package.json` seems like a good reason to make the cut.

Reply:
Ah, now I remember, thanks for the info.

So, yeah, not sure what the proper semver way would be. Perhaps release v1.0.0 final with the current engines, and then cut a new v2.0.0 with the bump in engines and tsconfig.